test-apps/sediment/src/main/java/com/twosigma/waiter/sediment/Servlet.java

```diff
@@ -0,0 +1,222 @@
+        final Response jettyResponse = (Response) response;
+
+        switch (request.getRequestURI()) {
+            case "/":
+            case "/status": {
+                handleStatusRequest(jettyRequest, jettyResponse);
+                break;
+            }
+            case "/trailers": {
+                handleTrailersRequest(jettyRequest, jettyResponse);
+                break;
+            }
+            default: {
+                LOGGER.severe(correlationId + " unsupported endpoint: " + request.getRequestURI());
+                super.service(request, response);
+                break;
+            }
+        }
+        LOGGER.info(correlationId + " done processing request");
+    }
+
+    void handleStatusRequest(final Request request, final Response response) throws IOException {
+        final String contentParam = request.getParameter("content");
+        final String responseContent = contentParam == null ? "OK" : contentParam;
+        response.setContentType("text/plain");
+        response.getWriter().write(responseContent);
+    }
+
+    void handleTrailersRequest(final Request request, final Response response) throws IOException {
+        final Map<String, Object> infoMap = populateRequestInfo(request);
+        final String responseBody = new Gson().toJson(infoMap);
+        final byte[] outputBytes = responseBody.getBytes();
+
+        final HttpFields trailer = retrieveTrailers(request);
+        if (trailer != null) {
+            response.setTrailers(() -> trailer);
+        }
+
+        final String responseStatusString = request.getHeader("x-sediment-response-status");
+        if (responseStatusString == null) {
+            response.setStatus(200);
+        } else {
+            response.setStatus(Integer.parseInt(responseStatusString));
+        }
+
+        final ByteArrayInputStream respInputStream = new ByteArrayInputStream(outputBytes);
+        final ServletOutputStream respOutputStream = response.getOutputStream();
+        while (true) {
+            final byte[] tempBytes = new byte[128];
+            final int bytesRead = respInputStream.read(tempBytes);
+            if (bytesRead == -1) {
+                final String sleepMsString = request.getHeader("x-sediment-sleep-before-response-trailer-ms");
+                if (sleepMsString != null) {
+                    try {
+                        LOGGER.info("sleeping " + sleepMsString + " ms before returning response trailers");
+                        Thread.sleep(Long.parseLong(sleepMsString));
+                    } catch (InterruptedException ex) {
+                        LOGGER.log(Level.SEVERE, "error while sleeping before sending trailers", ex);
+                    }
+                }
+                break;
+            }
+            respOutputStream.write(tempBytes, 0, bytesRead);
+            respOutputStream.flush();
+        }
+    }
+
+    private HttpFields retrieveTrailers(Request request) {
```

Comment:
Could we name this `createResponseTrailers` instead of `retrieveTrailers`?

Reply:
Done.